hedera-node/cli-clients/src/main/java/com/hedera/services/cli/sign/AccountBalanceSigningUtils.java

```
@@ -0,0 +1,124 @@
+        //        SettingsCommon.maxTransactionBytesPerEvent = Integer.MAX_VALUE;
+        //        SettingsCommon.transactionMaxBytes = Integer.MAX_VALUE;
+        //        SettingsCommon.maxAddressSizeAllowed = Integer.MAX_VALUE;
+    }
+
+    /**
+     * Generates a signature file for the account balance file
+     *
+     * @param signatureFileDestination the full path where the signature file will be generated
+     * @param streamFileToSign         the stream file to be signed
+     * @param keyPair                  the keyPair used for signing
+     * @return true if the signature file was generated successfully, false otherwise
+     */
+    public static boolean signAccountBalanceFile(
+            @NonNull final Path signatureFileDestination,
+            @NonNull final Path streamFileToSign,
+            @NonNull final KeyPair keyPair) {
+
+        Objects.requireNonNull(signatureFileDestination, "signatureFileDestination must not be null");
+        Objects.requireNonNull(streamFileToSign, "streamFileToSign must not be null");
+        Objects.requireNonNull(keyPair, "keyPair must not be null");
+
+        try {
+            final Hash entireHash = computeEntireHash(streamFileToSign.toFile());
+            final byte[] fileHashByte = entireHash.getValue();
+            final byte[] signature = sign(fileHashByte, keyPair);
+
+            generateSigBalanceFile(signatureFileDestination.toFile(), signature, fileHashByte);
+
+            System.out.println("Generated signature file: " + signatureFileDestination);
+
+            return true;
+        } catch (final SignatureException | IOException e) {
+            System.err.println("Failed to sign file " + streamFileToSign.getFileName() + ". Exception: " + e);
+            return false;
+        } catch (final InvalidKeyException | NoSuchAlgorithmException | NoSuchProviderException e) {
+            throw new RuntimeException("Irrecoverable error encountered", e);
```

Why throw here instead of `System.err` like in the previous catch clause?   (Especially since on a quick look this will fly off the top of the stack - anyway it isn't handled by caller `SignCommand.signFile`.)

---

Follow the platform's code style. Since it's RuntimeException, didn't have to catch it. @alittley

---

Yes but the question is: what happens to it?  Does it fly off the top of the stack and then the user sees a stack trace with some message buried somewhere in it?  Or is it handled by some caller and changed into a `System.err.println` that the user can understand?

---

You are right. We should not just throw exception. I will add System.err as I am working on the record stream sign.

---

Seems to not be changed yet.  Remember it for next time.

---

I fixed it and return false no one handle the throw that's why return false will be good.

---

@iwsimon followed the pattern @alittley is using an in case this exception happens it is more serious and we want to throw an exception with all the stuck traces. Any issue with that?